gradle/scripts/lib/common-dependencies.gradle

```diff
@@ -88,6 +88,64 @@ configure(dependencyManagementProject) {
     }
 }
 
+// Publish version catalog so that other projects can easily align transitive versions of the project by
+// importing the published catalog.
+// https://docs.gradle.org/current/userguide/platforms.html#sec:version-catalog-plugin
+configure(projectsWithFlag('version-catalogs')) { catalogProject ->
+    apply plugin: 'version-catalog'
+    apply plugin: 'maven-publish'
```

Comment:
Actually, I realized that the `publish` tag implies the install task and signing:

https://github.com/line/armeria/blob/26e57a871d496e7fc9e135bbf86225b74c708ab8/gradle/scripts/lib/common-publish.gradle#L66-L80

What do you think of making the user add a `publish` tag explicitly instead of implicitly adding this?

Reply:
Good point. `bom` and `version-catalogs` are implicitly associated with `publish`.
https://github.com/line/armeria/blob/a3690e8797e705612315bef96cd5ac489d6b90e1/gradle/scripts/lib/bom.gradle#L1-L13
But `pom.xml` is not correctly produced and singing is not supported.

> What do you think of making the user add a publish tag explicitly instead of implicitly adding this?

Agreed. The current documentation does not describe the behavior. It is necessary to set `publish` flag if users want to use the features.